### PR TITLE
Documentation and speedups to QuadExtFields impl of SqrtField

### PR DESCRIPTION
**Pull request by ValarDragon**

### CODE REVIEW

File: ff/src/fields/models/quadratic_extension.rs [hunk 1]
```diff
@@ -255,27 +260,44 @@ impl<'a, P: QuadExtParameters> SquareRootField for QuadExtField<P>
 where
     P::BaseField: SquareRootField,
 {
+    /// Outputs a LegendreSymbol, which indicates whether this field element is
+    ///  1 : a quadratic residue
+    ///  0 : equal to 0
+    /// -1 : a quadratic non-residue
```
Comment (Pratyush) [inline ff/src/fields/models/quadratic_extension.rs:266]: Maybe we move this to `SqrtField` trait itself?

File: ff/src/fields/models/quadratic_extension.rs [hunk 2]
```diff
@@ -82,6 +83,8 @@ pub trait QuadExtParameters: 'static + Send + Sync + Sized {
     }
 }
 
+/// An element of a quadratic extension field F_p[X]/(X^2 - NONRESIDUE) is
```
Comment (Pratyush) [inline ff/src/fields/models/quadratic_extension.rs:86]:
```suggestion
/// An element of a quadratic extension field F_p[X]/(X^2 - P::NONRESIDUE) is
```

File: ff/src/fields/models/quadratic_extension.rs [hunk 3]
```diff
@@ -82,6 +83,8 @@ pub trait QuadExtParameters: 'static + Send + Sync + Sized {
...
+/// represented as c0 + c1 * X, for c0, c1 in the base field.
```
Comment (Pratyush) [inline ff/src/fields/models/quadratic_extension.rs:87]:
```suggestion
/// represented as c0 + c1 * X, for c0, c1 in `P::BaseField`.
```

File: ff/src/fields/models/quadratic_extension.rs [hunk 4]
```diff
@@ -124,7 +127,9 @@ impl<P: QuadExtParameters> QuadExtField<P> {
         P::cyclotomic_exp(self, exponent)
     }
 
-    /// Norm of QuadExtField over P::BaseField: Norm(a) = a.x^2 - P::NON_RESIDUE * a.y^2
+    /// Norm of QuadExtField over P::BaseField: Norm(a) = a * a.conjugate()
```
Comment (Pratyush) [inline ff/src/fields/models/quadratic_extension.rs:130]:
```suggestion
    /// Norm of QuadExtField over `P::BaseField`: Norm(a) = `a * a.conjugate()`
```

File: ff/src/fields/models/quadratic_extension.rs [hunk 5]
```diff
@@ -124,7 +127,9 @@ impl<P: QuadExtParameters> QuadExtField<P> {
...
+    /// This simplifies to: Norm(a) = a.x^2 - P::NON_RESIDUE * a.y^2
```
Comment (Pratyush) [inline ff/src/fields/models/quadratic_extension.rs:131]:
```suggestion
    /// This simplifies to: `Norm(a) = a.x^2 - P::NON_RESIDUE * a.y^2`
```

File: ff/src/fields/models/quadratic_extension.rs [hunk 6]
```diff
@@ -124,7 +127,9 @@ impl<P: QuadExtParameters> QuadExtField<P> {
...
+    /// This is alternatively expressed as Norm(a) = a^(1 + p)
```
Comment (Pratyush) [inline ff/src/fields/models/quadratic_extension.rs:132]:
```suggestion
    /// This is alternatively expressed as `Norm(a) = a^(1 + p)`
```

File: ff/src/fields/models/quadratic_extension.rs [hunk 7]
```diff
@@ -255,27 +260,44 @@ impl<'a, P: QuadExtParameters> SquareRootField for QuadExtField<P>
 where
     P::BaseField: SquareRootField,
 {
+    /// Outputs a LegendreSymbol, which indicates whether this field element is
```
Comment (Pratyush) [inline ff/src/fields/models/quadratic_extension.rs:263]:
```suggestion
    /// Outputs a `LegendreSymbol`, which indicates whether this field element is
```

File: ff/src/fields/models/quadratic_extension.rs (hunk 1 above)
Comment (ValarDragon) [reply ff/src/fields/models/quadratic_extension.rs:266]: Good point, I'll do that

**Pull request merged**